src/server.c

```diff
@@ -6761,6 +6776,23 @@ int validateProcTitleTemplate(const char *template) {
     return ok;
 }
 
+int isMaxmemoryReservedLessThanMaxmemory(const char **err) {
+    if (server.maxmemory <= server.maxmemory_reserved) {
+        *err = "The maxmemory reserved value should be smaller than maxmemory.";
+        return C_ERR;
+    }
+    return C_OK;
+}
+
+void calculateKeyEvictionMemory(void) {
+    server.key_eviction_memory = server.maxmemory - server.maxmemory_reserved;
```

Comment:
`server.key_eviction_memory` is redundant information. Maybe we don't need to store it in a global? We can calculate `server.maxmemory - server.maxmemory_reserved` every time when we need it.

Reply:
I do not think so. In function getMaxmemoryState(), we need the server.key_eviction_memory value multiply times. And performEvictions() always call getMaxmemoryState() to check if key eviction process is needed.  Here, the server.maxmemory and server.maxmemory_reserved are fixed value if nobody update them by config set command.  Thus, I think calculating server.key_eviction_memory once and work as a global variable is better.